extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts

```diff
@@ -0,0 +1,401 @@
+			if (this._canceled) {
+				return;
+			}
+			//
+			azdata.tasks.startBackgroundOperation(
+				{
+					connection: undefined,
+					displayName: localize('mount.task.name', "Mounting HDFS folder on path {0}", this.props.hdfsPath),
+					description: '',
+					isCancelable: false,
+					operation: op => {
+						this.onSubmit(controller, op);
+					}
+				}
+			);
+		}
+	}
+
+	private async onSubmit(controller: ClusterController, op: azdata.BackgroundOperation): Promise<void> {
+		try {
+			await controller.mountHdfs(this.props.hdfsPath, this.props.remoteUri, this.credentials);
+			op.updateStatus(azdata.TaskStatus.InProgress, localize('mount.task.submitted', "Mount creation has started"));
+
+			// Wait until status has changed or some sensible time expired. If it goes over 2 minutes we assume it's "working"
+			// as there's no other API that'll give us this for now
+			let result = await this.waitOnMountStatusChange(controller);
+			let msg = result.state === MountState.Ready ? localize('mount.task.complete', "Mounting HDFS folder is complete")
+				: localize('mount.task.inprogress', "Mounting is likely to complete, check back later to verify");
+			op.updateStatus(azdata.TaskStatus.Succeeded, msg);
+		} catch (error) {
+			const errMsg = localize('mount.task.error', "Error mounting folder: {0}", (error instanceof Error ? error.message : error));
+			vscode.window.showErrorMessage(errMsg);
+			op.updateStatus(azdata.TaskStatus.Failed, errMsg);
+		}
+	}
+
+	private waitOnMountStatusChange(controller: ClusterController): Promise<MountInfo> {
+		return new Promise<MountInfo>((resolve, reject) => {
+			const waitTime = 5 * 1000; // 5 seconds
+			const maxRetries = 30;	// 5 x 30 = 150 seconds. After this time, can assume things are "working" as 2 min timeout passed
+			let waitOnChange = async (retries: number) => {
+				try {
+					let mountInfo = await this.getMountStatus(controller, this.props.hdfsPath);
+					if (mountInfo && mountInfo.error || mountInfo.state === MountState.Error) {
+						reject(new Error(mountInfo.error ? mountInfo.error : localize('mount.error.unknown', "Unknown error occurred during the mount process")));
+					} else if (mountInfo.state === MountState.Ready || retries <= 0) {
+						resolve(mountInfo);
+					} else {
+						setTimeout(() => {
+							waitOnChange(retries - 1).catch(e => reject(e));
+						}, waitTime);
+					}
+				} catch (err) {
+					reject(err);
+				}
+			};
+			waitOnChange(maxRetries);
+		});
+	}
+
+	private async getMountStatus(controller: ClusterController, path: string): Promise<MountInfo> {
+		let statusResponse = await controller.getMountStatus(path);
+		if (statusResponse.mount) {
+			return Array.isArray(statusResponse.mount) ? statusResponse.mount[0] : statusResponse.mount;
+		}
+		return undefined;
+	}
+}
+
+abstract class HdfsDialogBase<T extends DialogProperties> {
+
+	protected dialog: azdata.window.Dialog;
+	protected uiModelBuilder: azdata.ModelBuilder;
```

Comment:
looks like you don't have to do: protected uiModelBuilder!:azdata.ModelBuilder; Did you do something to avoid the tslint error of possible undefined for this property since it is not initialized in constructor?

Reply:
Extensions don't have the strict null checks run on them currently (but yeah we should still try to adhere by those rules so it isn't as painful to enable them in the future)

Reply:
Good learning for me, thanks! I'm adding the ! since these are not expected to be null (createDialog must be called)